File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcCurve/IfcBoundedCurve/IfcBSplineCurve/IfcBSplineCurve.vb
Public Class IfcBSplineCurve Inherits IfcBoundedCurve
    Public Degree As IfcInteger
    Public ControlPointsList As List(Of IfcCartesianPoint)
    Public CurveForm As IfcBSplineCurveForm
    Public ClosedCurve As IfcLogical
    Public SelfIntersect As IfcLogical

    ' === EXTENDED BY ===
    ' IfcBSplineCurveWithKnots

    ' === WHERE CLAUSES ===
    ' SameDim : SIZEOF(QUERY(Temp <* ControlPointsList | Temp.Dim <> ControlPointsList[1].Dim)) = 0

    ' === DERIVE CLAUSES ===
    ' UpperIndexOnControlPoints : IfcInteger := (SIZEOF(ControlPointsList) - 1)
    ' ControlPoints : ARRAY [0:UpperIndexOnControlPoints] OF IfcCartesianPoint := IfcListToArray(ControlPointsList,0,UpperIndexOnControlPoints)
End Class

File: src/main/resources/output/IFC4X3_java/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcCurve/IfcBoundedCurve/IfcBSplineCurve/IfcBSplineCurve.java
public class IfcBSplineCurve extends IfcBoundedCurve {
    public IfcInteger Degree;
    public List<IfcCartesianPoint> ControlPointsList;
    public IfcBSplineCurveForm CurveForm;
    public IfcLogical ClosedCurve;
    public IfcLogical SelfIntersect;

    // === EXTENDED BY ===
    // IfcBSplineCurveWithKnots

    // === WHERE CLAUSES ===
    // SameDim : SIZEOF(QUERY(Temp <* ControlPointsList | Temp.Dim <> ControlPointsList[1].Dim)) = 0

    // === DERIVE CLAUSES ===
    // UpperIndexOnControlPoints : IfcInteger := (SIZEOF(ControlPointsList) - 1)
    // ControlPoints : ARRAY [0:UpperIndexOnControlPoints] OF IfcCartesianPoint := IfcListToArray(ControlPointsList,0,UpperIndexOnControlPoints)
}
